### PR TITLE
Add proxy_wasm::main macro.

### PR DESCRIPTION
**Pull request by PiotrSikora**
This is primarly a mechanism to deliver a workaround for a breaking
change in Rust v1.56.0 (which updated LLVM to v13.0), which changed
existing reactors into multi-entry commands, leading to performance
degradation and/or unusable Proxy-Wasm plugins.

See: https://github.com/WebAssembly/WASI/issues/471

This is delivered as a macro to make it somehow compatible with the
unstable "-Z wasi-exec-model=reactor" feature.

Signed-off-by: Piotr Sikora <piotrsikora@google.com>

### CODE REVIEW

**Review by mathetake — APPROVED** [review]
intereseting

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -195,6 +195,41 @@ jobs:
...
+    - uses: actions/checkout@v2
+
+    - name: Update Rust
+      run: |
+        rustup toolchain install nightly --component clippy
+        rustup +nightly target add wasm32-wasi
+        rustup default nightly
+
+    - name: Rewrite Cargo.toml examples
+      run: grep -v '^crate-type' Cargo.toml > Cargo.tmp && mv Cargo.tmp Cargo.toml
```
Comment (Swaagie) [inline .github/workflows/rust.yml:211]:
```suggestion
      run: |
        grep -v '^crate-type' Cargo.toml > Cargo.tmp
        mv Cargo.tmp Cargo.toml
```

**Review by Swaagie — APPROVED** [review]

File: .github/workflows/rust.yml (hunk 1 above)
Comment (PiotrSikora) [reply .github/workflows/rust.yml]: Done.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,12 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## [Unreleased]
 
+### Fixed
+
+- Fixed performance degradation with `wasm32-wasi` target in Rust v1.56.0
+  or newer by adding `proxy_wasm::main` macro that should be used instead
+  of custom `_start`, `_initialize` and/or `main` exports.
```
Comment (PiotrSikora) [inline CHANGELOG.md:13]: @Swaagie @mathetake updated text to make it more clear that it's a fix and not simply a new feature. PTAL.

**Review by Swaagie — APPROVED** [review]
LGTM

**Pull request merged**